### PR TITLE
Clear out the microseconds when setting time.

### PR DESCRIPTION
**Pull request by markstory**
Refs #106

### CODE REVIEW

File: src/Traits/FrozenTimeTrait.php [hunk 1]
```diff
@@ -61,6 +61,10 @@ protected function stripTime($time)
      */
     public function setTime($hours, $minutes, $seconds = null, $microseconds = null)
     {
+        if (version_compare(PHP_VERSION, '7.1.0', '>=')) {
```
Comment (lorenzo) [inline src/Traits/FrozenTimeTrait.php]: can we make this a constant somewhere? Would like to avoid the comparison for each object created

Comment (markstory) [reply src/Traits/FrozenTimeTrait.php]: We could define a class/global constant inside of `src/carbon_compat.php`?

Comment (lorenzo) [reply src/Traits/FrozenTimeTrait.php]: Sounds good

**Comment by lorenzo** [comment]
Looks like we need to suppress a warning

**Pull request merged**